shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -656,6 +656,8 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
         .orElse(ctx.tyDefs.get(name).map(td => (td.kind, td.tparamsargs.size)))
         .toRight(() => err("type identifier not found: " + name, loc)(raise))
     val localVars = mutable.Map.empty[TypeVar, TypeVariable]
+    def tp(loco: Opt[Loc], desc: Str, originName: Opt[Str] = N) =
```

Comment:
I find it weird to see tp being redefined here considering it's the name we give to TypeProvenance.apply l.119.

Reply:
You're right that it was confusing! I've changed it to `tyTp`.